src/java/org/apache/cassandra/db/marshal/CompositeType.java

```diff
@@ -389,4 +423,28 @@ public String toString()
         out.flip();
         return accessor.valueOf(out);
     }
+
+    public static ByteBuffer build(boolean isStatic, ByteBuffer[] buffers, byte lastEoc)
```

Comment:
Can we add a "used for tests" comment or `@VisibleForTests`?

Reply:
This method is actually not used at all in this code - as I can see in DSE, it is used by the code path originating from `fromComparableBytes` which in turn is used by tries implementation in MemTables - am I right? If so, should I port `fromComparableBytes` as a part of this PR (or as another PR against the same branch) or It will be somewhere else, around Tries in MemTables implementation?

Reply:
On one hand, it's better to keep the PR shorter. On the other, the reverse construction is a part of the byte order and it would be great to have it to ease understanding.

Maybe the best combination is to leave it out from this PR, but have it available for consulting in a separate one (even if it's for another CEP) when we publish it?

Note: we have to do some coordination, because Dimitar is starting work on memtable tries as part of cooperation we have with Intel, and we don't want to repeat any of the work -- I will schedule a meeting for that.

Reply:
> Maybe the best combination is to leave it out from this PR, but have it available for consulting in a separate one (even if it's for another CEP) when we publish it?

that sounds good to me

> Note: we have to do some coordination, because Dimitar is starting work on memtable tries as part of cooperation we have with Intel, and we don't want to repeat any of the work

i'll try to create that separate PR ASAP so that Dimitar can use it